### PR TITLE
Add Markup plugin for Chartkit

### PR DESCRIPTION
**Pull request by artemipanchuk**

### CODE REVIEW

File: src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import './MarkupWidget.scss';
+
+const b = block('chartkit-markup');
+
+// eslint-disable-next-line react/display-name
+const MarkupWidget = React.forwardRef<ChartKitWidgetRef | undefined, MarkupWidgetProps>(
+    // _ref needs to avoid this React warning:
+    // "forwardRef render functions accept exactly two parameters: props and ref"
+    (props, _ref) => {
```
Comment (korvin89) [inline src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx:20]: U can add name here and remove `eslint-disable-next-line react/display-name`

File: src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            data: {data},
+        } = props;
+
+        const generatedId = React.useMemo(() => `${id}_${getRandomCKId()}`, [data, id]);
+
+        Performance.mark(generatedId);
+
+        React.useLayoutEffect(() => {
+            // TODO: swap to onRender after https://github.com/gravity-ui/chartkit/issues/33
+            onLoad?.({widgetRendering: Performance.getDuration(generatedId)});
```
Comment (korvin89) [inline src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx:33]: Lets use `onRender` here

File: src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            <div className={b()}>
+                <div className={b('wrapper')}>
+                    <Markup
+                        item={data}
+                        externalProps={{
+                            url: {
+                                onClick: (event: React.SyntheticEvent) => {
+                                    // need to stop propagation for link components because it works incorrect with sorting by rows
+                                    // user click by link it leads to call both actions at the same time
+                                    // now clicking on the link will only open it without sorting table
```
Comment (korvin89) [inline src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx:52]: Is relevant for using in ChartEditor only case?

File: src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        if (!data) {
+            throw new ChartKitError({
+                code: CHARTKIT_ERROR_CODE.NO_DATA,
+            });
+        }
+
+        return (
+            <div className={b()}>
+                <div className={b('wrapper')}>
```
Comment (korvin89) [inline src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/renderer/MarkupWidget.tsx:44]: Why we need this extra wrap?

File: src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/types.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import {ChartKitOnLoadData} from '@gravity-ui/chartkit';
+
+import {MarkupItem} from '../../../../../components/Markup';
+
+export type MarkupWidgetData = {
+    data: MarkupItem;
+};
+
+export type MarkupWidgetProps = {
```
Comment (korvin89) [inline src/ui/libs/DatalensChartkit/ChartKit/plugins/Markup/types.ts:9]: U can write this type as `ChartKitProps<'markup'>`

File: src/ui/libs/DatalensChartkit/types/widget.ts [hunk 6]
```diff
@@ -298,6 +298,11 @@ export interface YMapWidget extends WidgetBaseWithData {
     type: 'ymap';
 }
 
+export type MarkupWidget = WidgetBaseWithData & {
+    type: 'markup';
+    data: {};
```
Comment (korvin89) [inline src/ui/libs/DatalensChartkit/types/widget.ts:303]: Why empty object?

**Review by melikhov-dev — DISMISSED** [review]
By the way — server code is ok

**Review by korvin89 — APPROVED** [review]

**Comment by artemipanchuk** [comment]
Related to https://github.com/datalens-tech/datalens/issues/92

**Pull request merged**